docs/architecture/product_modelling/imports.md

```diff
@@ -0,0 +1,99 @@
+# Importing Existing Subscriptions
+When adding new products to Workflow Orchestrator, these products can rely on existing estate in the network that first
+needs to be added to the service database. This page describes a procedure that can be used to have a separate creation
+and import workflow for one product.
+
+As a requirement, each product type must have exactly one workflow where `target=Target.CREATE`. It is therefore not
+possible to have for example two workflows `create_node` and `import_node`. It is possible however, to have a separate
+product type, that relies on the same product block.
+
+## Using Separate Product Types
+In the code example below, an example is given for a `Node` product type that needs to be imported into Orchestrator.
+
+```python
+class Node(NodeProvisioning, lifecycle=[SubscriptionLifecycle.ACTIVE]):
+    """A node that is currently active."""
+    node: NodeBlock
+
+class ImportedNode(ImportedNodeInactive, lifecycle=[SubscriptionLifecycle.ACTIVE]):
+    """An imported node that is currently active."""
+    node: NodeBlock
+```
+
+In this example, both types contain the same product block, which is required for this approach to import existing
+products. The creation workflow for a `Node` will remain `create_node`, which is most likely a workflow that takes user
+input from a form page, and then interacts with external systems to provision a new node.
+
+For the `ImportedNode`, the creation workflow is `create_imported_node`. This workflow can be designed to still take
+user input from a form page, but could also be provided programmatically by either a CLI command or API endpoint that is
+called. Data sources can include external API resources, CSV- or YAML files, etc. If desired, interaction with all
+external provisioning systems can be skipped, resulting in a creation workflow that could be as simple as follows.
+
+```python
+from orchestrator.workflow import StepList, begin, done, workflow
+from orchestrator.workflows.steps import resync, store_process_subscription
+
+@workflow("Create imported Node", target=Target.CREATE)
+def create_imported_node() -> StepList:
```

Comment:
Perhaps you already know and intentionally didn't use them, but `orchestrator.workflows.utils` contains wrappers `@create_workflow` and `@modify_workflow` that simplify some of the boilerplate

Reply:
I've seen them, yes, but actually forgot to use them. I'd like to do that, even if it's just as an exercise to see whether I can use them correctly. 🙂